appserver/security/webintegration/src/main/java/com/sun/web/security/RealmAdapter.java

```diff
@@ -1200,25 +1029,26 @@ protected Principal getPrincipal(String username) {
 
     // START OF IASRI 4809144
     /**
-     * This method is added to create a Principal based on the username only. Hercules stores the username as part of authentication
-     * failover and needs to create a Principal based on username only <sridhar.satuloori@sun.com>
+     * This method is added to create a Principal based on the username only. Hercules stores the username as part of
+     * authentication failover and needs to create a Principal based on username only <sridhar.satuloori@sun.com>
      *
      * @param username
      * @return Principal for the user username HERCULES:add
      */
     public Principal createFailOveredPrincipal(String username) {
-        _logger.log(Level.FINEST, "IN createFailOveredPrincipal (" + username + ")");
-        // set the appropriate security context
+        _logger.log(FINEST, () -> "IN createFailOveredPrincipal (" + username + ")");
         loginForRunAs(username);
-        SecurityContext secCtx = SecurityContext.getCurrent();
-        _logger.log(FINE, "Security context is " + secCtx);
-        assert (secCtx != null);
-        Principal principal = new WebPrincipal(username, (char[]) null, secCtx);
-        _logger.log(Level.INFO, "Principal created for FailOvered user " + principal);
+
+        // set the appropriate security context
+        SecurityContext securityContext = SecurityContext.getCurrent();
+        _logger.log(FINE, () -> "Security context is " + securityContext);
+
+        Principal principal = new WebPrincipal(username, (char[]) null, securityContext);
+        _logger.log(INFO, () -> "Principal created for FailOvered user " + principal);
```

Comment:
Some time ago in Payara I made some benchmark and to my surprise i found that suppliers are not faster. Maybe I should try it again with JDK17 ...